Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -0,0 +1,287 @@
+				stepper.ConstraintTo (numericEditor, (s, n) => s.Left == n.Right + 5),
+				stepper.ConstraintTo (numericEditor, (s, n) => s.Top == n.Top + 1),
+			});
+		}
+
+		protected virtual void SetStepperActivated ()
+		{
+			SetValue (stepper.DoubleValue);
+		}
+
+		protected void OnStepperActivated (object sender, EventArgs e)
+		{
+			if (!editing) {
+				editing = true;
+				SetStepperActivated ();
+				OnStepperValueChanged ();
+				if (ValueChanged != null)
+					ValueChanged (this, EventArgs.Empty);
+				editing = false;
+			}
+		}
+
+		protected virtual void OnStepperValueChanged ()
+		{
+			if (StepperValueChanged != null)
+				StepperValueChanged (this, EventArgs.Empty);
+		}
+
+		protected void OnFieldEditingEnded (object sender, EventArgs e)
+		{
+			if (!editing) {
+				editing = true;
+				SetValue (numericEditor.StringValue);
+				if (FieldEditingEnded != null)
+					FieldEditingEnded (this, EventArgs.Empty);
+				if (ValueChanged != null)
+					ValueChanged (this, EventArgs.Empty);
+				editing = false;
+			}
+		}
+
+		public virtual void SetValue (string value)
+		{
+			//Regulates maximun and minium out of range
+			stepper.DoubleValue = CoerceValue (FieldValidation.FixInitialValue (value, Value.ToEditorString ()).ToEditorDouble ());
+			numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
+		}
+
+		public virtual void SetValue (double value)
+		{
+			//Regulates maximun and minium out of range
+			stepper.DoubleValue = CoerceValue (value);
+			numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
+		}
+
+		protected double CoerceValue (double val)
+		{
+			return FieldValidation.CoerceValue (val, MinimumValue, MaximumValue);
+		}
+
+		public void AddAccessoryView (NSControl control)
+		{
+			AddSubview (control);
+			AddConstraints (new NSLayoutConstraint[] {
+				NSLayoutConstraint.Create (control, NSLayoutAttribute.CenterY, NSLayoutRelation.Equal, this, NSLayoutAttribute.CenterY, 1, 1),
+				NSLayoutConstraint.Create (control, NSLayoutAttribute.Right, NSLayoutRelation.Equal, this, NSLayoutAttribute.Right, 1, 0),
+				//NSLayoutConstraint.Create (control, NSLayoutAttribute.Width, NSLayoutRelation.GreaterThanOrEqual, null, NSLayoutAttribute.NoAttribute, 1, 80),
+				NSLayoutConstraint.Create (stepper, NSLayoutAttribute.Right, NSLayoutRelation.Equal, control, NSLayoutAttribute.Left, 1, -6)
+			});
+			// tODO RemoveConstraint (rightSideConstraint);
+		}
+
+		public void SimulateTextInput (string input)
+		{
+			numericEditor.StringValue = input;
+			OnFieldEditingEnded (numericEditor, EventArgs.Empty);
+		}
+
+		public void GoUp ()
+		{
+			stepper.DoubleValue = stepper.DoubleValue + IncrementValue;
+			OnStepperActivated (stepper, EventArgs.Empty);
```

Comment:
We should ensure the `Stepper` value and `NumericTextField` value are in sync before any events visible outside of this class are invoked, otherwise people could see inconsistent values!